velox/functions/sparksql/DecimalArithmetic.cpp

```diff
@@ -253,10 +254,12 @@ struct DecimalAddSubtractBase {
   }
 
   // Computes the result precision and scale for decimal add and subtract
-  // operations following Hive's formulas.
+  // operations following Hive's formulas when `allowPrecisionLoss` is true.
   // If result is representable with long decimal, the result
   // scale is the maximum of 'aScale' and 'bScale'. If not, reduces result scale
```

Comment:
The difference is only on whether 'adjustPrecisionScale' or 'bounded' is called, so we need to move the change on L257 perhaps before 'If not'.